sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -872,6 +872,10 @@ public partial class HttpClientTransport : Azure.Core.Pipeline.HttpPipelineTrans
         public override void Process(Azure.Core.HttpMessage message) { }
         public override System.Threading.Tasks.ValueTask ProcessAsync(Azure.Core.HttpMessage message) { throw null; }
     }
+    public static partial class HttpMessageOptions
+    {
+        public static void SetTelemetryPackage(Azure.Core.HttpMessage message, Azure.Core.Pipeline.TelemetryPackageInfo packageInfo) { }
```

Comment:
I think there was a misunderstanding. I think TelemetryPackageInfo should be called HttpMessageOptions. The static class should be called HttpMessageExtensions, like you had it before.

Reply:
Are you suggesting it the method would look like this?:

```c#
public static void SetOptions(this HttpMessage message, HttpMessageOptions options)
```

I don't think `HttpMessageOptions` is specific enough to describe its usage. My thinking was that if a Type is going to be added to the dictionary, it be focused on a specific use case.

The change away from the extension method was to distinguish this from an instance method on HttpMessage itself. Assuming it should be an instance method, would it make more sense to keep the method defined directly on HttpMessage?

Reply:
What about 
```csharp
public static class HttpMessageExtensions {
    public static void SetTelemetryOptions(this HttpMessage message, TelemetryOptions options)
}
```